src/vs/workbench/contrib/comments/browser/commentsView.ts

```diff
@@ -131,13 +135,10 @@ export class CommentsPanel extends ViewPane {
 		await this.tree.setInput(this.commentsModel);
 	}
 
-	public getActions(): IAction[] {
-		if (!this.collapseAllAction) {
-			this.collapseAllAction = new Action('vs.tree.collapse', nls.localize('collapseAll', "Collapse All"), 'collapse-all', true, () => this.tree ? new CollapseAllAction<any, any>(this.tree, true).run() : Promise.resolve());
-			this._register(this.collapseAllAction);
+	public collapseAll() {
+		if (this.tree) {
+			new CollapseAllAction<any, any>(this.tree, true).run();
```

Comment:
just a suggestion, but I think it would be simpler to put

```ts
this.tree.collapseAll();
this.tree.setSelection([]);
this.tree.setFocus([]);
this.tree.domFocus();
this.tree.focusFirst()
```

here instead of creating a new action each time

Reply:
Yeah makes sense. Will inline them after merging.